web/src/compose.js

```diff
@@ -169,7 +178,137 @@ export function send_message_success(request, data) {
     }
 }
 
-export function send_message(request = create_message_object()) {
+let split_long_messages_banner_shown = false;
+
+export function update_split_long_messages_info_banner() {
+    if (check_need_to_split_long_messages()) {
+        show_split_long_messages_info_banner();
+        compose_ui.autosize_textarea($("textarea#compose-textarea"));
+    } else {
+        clear_split_long_messages_info_banner();
+    }
+}
+
+const SPLIT_DELIMITER = "\n\n\n";
+/**
+ * checks whether the message actually needs to be split, as it is possible that `split_long_messages` is enabled but the message doesn't contain the delimiter.
+ */
+function check_need_to_split_long_messages() {
+    if (!is_split_long_messages_enabled()) {
+        return false;
+    }
+    const content = compose_state.message_content();
+    return content.includes(SPLIT_DELIMITER);
+}
+
+export function is_split_long_messages_enabled() {
+    return localStorage.getItem("split_long_messages") === "true";
+}
+
+function show_split_long_messages_info_banner() {
+    if (split_long_messages_banner_shown) {
+        return;
+    }
+    const context = {
+        banner_type: compose_banner.INFO,
+        banner_text: $t({
+            defaultMessage: `Your message will be split into different parts.`,
```

Comment:
How did you pick this string? It doesn't seem to be the one in the issue description.

Also, please don't use backticks for strings containing no syntax.

Reply:
The issue specifies "The text in you compose box will be split at two consecutive blank lines into N messages."
But counting the `N` seemed expensive in terms of performance, so I had left it like that for the time being. This is also why there are backticks.